### PR TITLE
[PHOTO-1087] fix buildtime warnings

### PR DESCRIPTION
**Pull request by joemuers**
This PR resolves the compiler warnings (for PhotoLeap, at least) that we are getting from this library.

Changes:
- Updates deployment target to iOS 12.0.
- Adds "English" to list of known localisations.
- Resolves the new warnings that came up (no actual logic is changed in the execution).  Note that I had to suppress deprecations warnings entirely for `YapDatabaseCloudCorePipeline.m`.
- Updates Xcode last version check in the scheme to 1340.

See https://lightricks.atlassian.net/browse/PHOTO-1087 for more details.

### CODE REVIEW

File: YapDatabase/Extensions/CloudKit/Utilities/YDBCKChangeSet.m [hunk 1]
```diff
@@ -164,15 +164,15 @@ - (NSUInteger)recordsToSaveCount
...
+		return [NSKeyedArchiver archivedDataWithRootObject:deletedRecordIDs requiringSecureCoding:NO error:nil];
 	else
 		return nil;
 }
 
 - (void)deserializeDeletedRecordIDs:(NSData *)serializedDeletedRecordIDs
 {
 	if (serializedDeletedRecordIDs)
-		deletedRecordIDs = [NSKeyedUnarchiver unarchiveObjectWithData:serializedDeletedRecordIDs];
+    deletedRecordIDs = [NSKeyedUnarchiver unarchivedObjectOfClass:[NSMutableArray class] fromData:serializedDeletedRecordIDs error:nil];
```
Comment (cameron-lightricks) [inline YapDatabase/Extensions/CloudKit/Utilities/YDBCKChangeSet.m:175]: indent fix here

File: YapDatabase/Extensions/CloudKit/Utilities/YDBCKChangeSet.m [hunk 2]
```diff
@@ -186,15 +186,15 @@ - (NSData *)serializeModifiedRecords
...
+		return [NSKeyedArchiver archivedDataWithRootObject:[modifiedRecords allValues] requiringSecureCoding:NO error:nil];
 }
 
 - (void)deserializeModifiedRecords:(NSData *)serializedModifiedRecords
 {
 	NSArray *modifiedRecordsArray = nil;
 	
 	if (serializedModifiedRecords) {
-		modifiedRecordsArray = [NSKeyedUnarchiver unarchiveObjectWithData:serializedModifiedRecords];
+    modifiedRecordsArray = [NSKeyedUnarchiver unarchivedObjectOfClass:[NSArray class] fromData:serializedModifiedRecords error:nil];
```
Comment (cameron-lightricks) [inline YapDatabase/Extensions/CloudKit/Utilities/YDBCKChangeSet.m:197]: indent fix here

File: YapDatabase/YapDatabase.m [hunk 3]
```diff
@@ -158,7 +158,10 @@ + (YapDatabaseSerializer)defaultSerializer
 + (YapDatabaseDeserializer)defaultDeserializer
 {
 	return ^ id (NSString __unused *collection, NSString __unused *key, NSData *data){
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
 		return data && data.length > 0 ? [NSKeyedUnarchiver unarchiveObjectWithData:data] : nil;
+#pragma clang diagnostic pop
```
Comment (cameron-lightricks) [inline YapDatabase/YapDatabase.m:164]: Is this to get around the fact that we do not know the type of the class at deserialisation time?

Comment (cameron-lightricks) [reply YapDatabase/YapDatabase.m:164]: Annoyingly `unarchiveObjectWithData:` doesn't exist beyond iOS 12, so we'll need to actually change this code soon, if we cannot drop YapDatabase :(

File: YapDatabase/YapDatabase.m [hunk 4]
```diff
@@ -231,7 +234,10 @@ + (YapDatabaseDeserializer)timestampDeserializer
 		}
 		else
 		{
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
 			return [NSKeyedUnarchiver unarchiveObjectWithData:data];
+#pragma clang diagnostic pop
```
Comment (cameron-lightricks) [inline YapDatabase/YapDatabase.m:240]: I'm guessing this can't be updated for the same reasons as above?

**Review by cameron-lightricks — APPROVED** [review]
Approved, with a few nitpicky questions and indent fixes 🚀

File: YapDatabase/YapDatabase.m (hunk 3 above)
Comment (joemuers) [reply YapDatabase/YapDatabase.m:164]: That's right.
Probably not super hard to add the types if we really do want to keep YapDB. But I didn't want to waste time doing it here.

File: YapDatabase/YapDatabase.m (hunk 4 above)
Comment (joemuers) [reply YapDatabase/YapDatabase.m:240]: Yep.

**Pull request merged**